src/app/business/block/bucket-detail/bucket-detail.component.ts

```diff
@@ -101,6 +105,76 @@ export class BucketDetailComponent implements OnInit {
       this.getTypes();
     }
     );
+    this.copySelectedDir = window.sessionStorage['searchIndex'] != "" ? JSON.parse(window.sessionStorage.getItem("searchIndex")) : [];
+  }
+  clickOperate(){
+    if(this.selectedDir.length >0){
+      this.isReadyCopy = false;
+      this.copySelectedDir = lodash.cloneDeep(this.selectedDir);
+    }else{
+      this.isReadyCopy = true;
+      this.copySelectedDir = window.sessionStorage['searchIndex'] != "" ? JSON.parse(window.sessionStorage.getItem("searchIndex")) : [];
+      this.isReadyPast = this.copySelectedDir.length != 0 ? false :true;
+    }
+    this.moreItems = [
+      {
+        disabled: this.isReadyCopy,
+        label: 'copy', command: () => {
+          this.copyObject();
+        }
+      },
+      {
+        disabled: this.isReadyPast,
+        label: 'paste', command: (y) => {
+          this.pasteObject();
+        }
+      },
+    ];
+  }
+  //copy object
+  copyObject() {
+    this.isReadyPast = true;
+    this.copySelectedDir.forEach((item, index) =>{
+      this.copySelectedDir[index]['source'] = this.bucketId;
+      this.copySelectedDir[index]['folderId'] = this.folderId;
+    })
+    window.sessionStorage['searchIndex'] = JSON.stringify(this.copySelectedDir);
+  }
+  //paste object
+  pasteObject() {
+    this.copySelectedDir.forEach(item=>{
+      let key = this.folderId != "" ? this.folderId + item.Key : item.Key;
```

Comment:
Is this item.Key or item.Contents.Key ? Please check.

Reply:
thank you ,I have made the change.